packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -242,6 +239,50 @@ export class DataProxyEngine extends Engine {
     return [host, apiKey]
   }
 
+  private mergeOverriddenDatasources(): Record<string, InlineDatasource> {
+    if (this.config.datasources === undefined) {
+      return this.inlineDatasources
+    }
+
+    const finalDatasources = klona(this.inlineDatasources)
```

Comment:
Can you please check how much this increases the size of the edge bundle (if anything)?

Reply:
(something to keep watching because it can easily explode. It's 263KB as of 4.1.0)

Reply:
I simplified code based on your other comment so that the deep copy is not necessary anymore.